woocommerce/woocommerce
# Rework confusing shipping message on cart page when calculator is disabled and no methods are available.

### All Submissions:


### Changes proposed in this Pull Request:

Handles edge cases where users get message asking to _"Enter the address."_ on cart page when shipping calculator is disabled and no shipping methods are available. There is also another edge case where default customer location is set to no location by default and shipping zone for rest of the locations has no shipping methods.

These are sort of badly configured stores but because these related settings are on three different tabs this ought to happen.

Whereas it should convey that shipping can be updated on checkout as we don't have all the information to determine if shipping is actually available or not for the destination.

Closes #24674 .

### How to test the changes in this Pull Request:

1. Disable Shipping Calculator
2. Set shipping zone for other than your geolocation & disable any methods for _Locations not covered by your other zones_
3. Add products to cart & visit cart page
4. You should see _'Shipping options will be updated at checkout.'_

### Other information:




### Changelog entry

> Fix issue where if shipping calculator is disabled and no shipping methods are available then customer gets confusing message on cart page.
